crates/environ/src/compile/module_types.rs

```diff
@@ -163,7 +163,10 @@ impl ModuleTypesBuilder {
                     let idx = self.types.push(WasmSubType {
                         is_final: true,
                         supertype: None,
-                        composite_type: WasmCompositeType::Func(f.clone()),
+                        composite_type: WasmCompositeType {
+                            inner: WasmCompositeInnerType::Func(f.clone()),
+                            shared: false, // TODO: handle shared
```

Comment:
Here I think you'll want to propagate the shared-ness from `self.types[for_func_ty]` up above